### ISSUE TITLE
State based TextureButton  color modulation  

### ISSUE THREAD

**Issue by LordBrkica** [godotengine/godot-proposals#2856]
### Describe the project you are working on
Game

### Describe the problem or limitation you are having in your project
I have a need for texture button with different colors/modulation depending on state.

Disabled - transparency 0.5
Pressed - bluish
Normal - normal

Using modulate only changes the normal state.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation
The "normal" button has one brilliant feature: Custom Colors. One can change the colors depending on the button state, so we have:
Font Color Disabled
Font Color
Font Color Hovered
Font Color Pressed

It would be greate to have the same option in the Texture button !

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams
It would work the same as the fonts colors work. It would be modulate but for different states.

### If this enhancement will not be used often, can it be worked around with a few lines of script?
Almost everything can be worked around...

Right now I have a 3 .pngs with (3 different colors) and use that as 3 different textures in texture button.

It would be great if texture button could have the ability to modulate the different textures.
### Is there a reason why this should be core and not an add-on in the asset library?
If we support different colors at normal button I thinks its reasonable to support the same for textures.

**Comment by dalexeev** [comment]
Related:  #2498.

**Comment by Calinou** [comment]
As a workaround, you can use a Button with a custom StyleBoxTexture which has a Modulate property.

**Comment by LordBrkica** [comment]
> As a workaround, you can use a Button with a custom StyleBoxTexture which has a `modulate_color` property. You will need to create one StyleBoxTexture per state, but you can reuse the same texture across all states.

Thats true and is probably the workaround closest to my need. But StyleBoxTexture lacks the Stretch Mode property (and several styles there defined) a TextureButton has.
